services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java

```diff
@@ -84,13 +87,42 @@ public boolean isValid(SubmissionPayload submissionPayload, ConstraintValidatorC
       List<TemporaryExposureKey> exposureKeys = submissionPayload.getKeysList();
       validatorContext.disableDefaultConstraintViolation();
 
-      return checkStartIntervalNumberIsAtMidNight(exposureKeys, validatorContext)
+      boolean isValidPayload = checkStartIntervalNumberIsAtMidNight(exposureKeys, validatorContext)
           && checkKeyCollectionSize(exposureKeys, validatorContext)
           && checkOriginCountryIsValid(submissionPayload, validatorContext)
           && checkVisitedCountriesAreValid(submissionPayload, validatorContext)
           && checkRequiredFieldsNotMissing(exposureKeys, validatorContext)
           && checkTransmissionRiskLevelIsAcceptable(exposureKeys, validatorContext)
           && checkDaysSinceOnsetOfSymptomsIsInRange(exposureKeys, validatorContext);
+
+      if (!isValidPayload) {
+        String payloadString = generatePrintablePayload(submissionPayload);
+        logger.error("Errors caused by invalid payload {}", payloadString);
+      }
+      return isValidPayload;
+    }
+
+    private String generatePrintablePayload(SubmissionPayload submissionPayload) {
```

Comment:
The solution is good enough if we need it quickly, but one suggestion from my side is to extract this sort of "produce printable object" logic outside the validator in a dedicated logging utility class. This would encapsulate such printing responsabilities and can be further extended with other objects.  You can then have unit tests around how the printed object should look like (i.e. not contain sensitive fields maybe)